bootstrap.js

```diff
@@ -130,41 +130,58 @@ async function install() {
 }
 
 // eslint-disable-next-line no-unused-vars
-async function uninstall(aData, aReason) {
-  if (aReason === ADDON_UNINSTALL
-      || aReason === ADDON_DISABLE) {
-    const config = await getConfig();
-    const storedPrefs = config.savedConfiguration.prefs || {};
-    PREFS.forEach((pref) => {
-      let value = pref.default;
-      if (pref.name in storedPrefs) {
-        value = storedPrefs[pref.name];
-      }
-      if ("int" === pref.type) {
-        Services.prefs.setIntPref(pref.name, value);
-      } else {
-        Services.prefs.setBoolPref(pref.name, value);
-      }
-    });
+async function uninstall({resourceURI}, aReason) {
+  if (checkLegacyFirefox()) {
+    if (aReason === ADDON_UNINSTALL) {
+      unloadStyles(resourceURI);
+      await removeChanges();
+    }
   }
 }
 
-// eslint-disable-next-line no-unused-vars
-function startup({webExtension, resourceURI}) {
+async function removeChanges() {
+  const config = await getConfig();
+  const storedPrefs = config.savedConfiguration.prefs || {};
+  PREFS.forEach((pref) => {
+    let value = pref.default;
+    if (pref.name in storedPrefs) {
+      value = storedPrefs[pref.name];
+    }
+    if ("int" === pref.type) {
+      Services.prefs.setIntPref(pref.name, value);
+    } else {
+      Services.prefs.setBoolPref(pref.name, value);
+    }
+  });
+}
+
+function checkLegacyFirefox() {
   const version = Services.appinfo.version;
   const versionMatch = version.match(/^([0-9]+)\./)[1];
-  if (versionMatch === "55"
-      || versionMatch === "56") {
+  if (Number(versionMatch) <= 56) {
+    return false;
```

Comment:
This logic seems backwards, or the function name is confusing. "LegacyFirefox" would be <= 56?

Reply:
Yeah sorry I renamed last minute, will work on this tomorrow.